source/common/http/filter/gzip_filter.cc

```diff
@@ -0,0 +1,282 @@
+  if (accept_encoding) {
+    bool is_identity{true};  // true if does not exist or is followed by `q=0`.
+    bool is_wildcard{false}; // true if found and not followed by `q=0`.
+    bool is_gzip{false};     // true if exists and not followed by `q=0`.
+
+    std::vector<absl::string_view> tokens =
+        StringViewUtil::splitToken(headers.AcceptEncoding()->value().c_str(), ";,", false);
+
+    for (auto& token : tokens) {
+      absl::string_view next_token{};
+      absl::string_view trimmed_token = StringViewUtil::trim(token);
+      if (token != tokens.back()) {
+        next_token = StringViewUtil::trim(*(&token + 1));
+      }
+      if (trimmed_token == Http::Headers::get().AcceptEncodingValues.Gzip) {
+        is_gzip = next_token.empty() ? true : (next_token != ZeroQvalueString);
+      }
+      if (trimmed_token == Http::Headers::get().AcceptEncodingValues.Identity) {
+        is_identity = next_token.empty() ? false : (next_token == ZeroQvalueString);
+      }
+      if (trimmed_token == Http::Headers::get().AcceptEncodingValues.Wildcard) {
+        is_wildcard = !next_token.empty() ? true : (next_token != ZeroQvalueString);
+      }
+    }
+
+    return ((is_gzip && is_identity) || is_wildcard);
+  }
+
+  return true;
+}
+
+bool GzipFilter::isCacheControlAllowed(HeaderMap& headers) const {
+  const Http::HeaderEntry* cache_control = headers.CacheControl();
+  if (cache_control) {
+    return !StringViewUtil::findToken(cache_control->value().c_str(), ";=",
+                                      Http::Headers::get().CacheControlValues.NoTransform.c_str());
+  }
+  return true;
+}
+
+bool GzipFilter::isContentTypeAllowed(HeaderMap& headers) const {
+  const Http::HeaderEntry* content_type = headers.ContentType();
+  if (content_type && !config_->contentTypeValues().empty()) {
+    // TODO(gsagula): do this lookup without allocating a new string.
+    std::string value{StringViewUtil::cropRight(content_type->value().c_str(), ";")};
+    return config_->contentTypeValues().find(value) != config_->contentTypeValues().end();
+  }
+  return true;
+}
+
+bool GzipFilter::isEtagAllowed(HeaderMap& headers) const {
+  const Http::HeaderEntry* etag = headers.Etag();
+  if (config_->disableOnEtag()) {
+    return !etag;
+  }
+  if (etag) {
+    absl::string_view value(etag->value().c_str());
+    if (!(value.length() > 2 && (value[0] == 'w' || value[0] == 'W') && value[1] == '/')) {
+      headers.removeEtag();
+    }
+  }
+  return true;
+}
+
+bool GzipFilter::isLastModifiedAllowed(HeaderMap& headers) const {
+  return !(config_->disableOnLastModified() && headers.LastModified());
+}
+
+bool GzipFilter::isMinimumContentLength(HeaderMap& headers) const {
+  const Http::HeaderEntry* content_length = headers.ContentLength();
+  if (content_length) {
+    uint64_t length;
+    return StringUtil::atoul(content_length->value().c_str(), length)
+               ? length >= config_->minimumLength()
```

Comment:
You could use &&
  return StringUtil::atoul(content_length->value().c_str(), length) && length >= config_->minimumLength();

Reply:
👍